src/Lti/Service/GetUserAssignmentLtiRequestService.php

```diff
@@ -22,102 +22,29 @@
 
 namespace OAT\SimpleRoster\Lti\Service;
 
-use Carbon\Carbon;
-use JsonException;
 use OAT\SimpleRoster\Entity\Assignment;
-use OAT\SimpleRoster\Entity\LineItem;
-use OAT\SimpleRoster\Entity\LtiInstance;
 use OAT\SimpleRoster\Exception\AssignmentNotProcessableException;
-use OAT\SimpleRoster\Generator\NonceGenerator;
-use OAT\SimpleRoster\Lti\LoadBalancer\LtiInstanceLoadBalancerInterface;
+use OAT\SimpleRoster\Lti\Factory\LtiRequestFactoryInterface;
 use OAT\SimpleRoster\Lti\Request\LtiRequest;
-use OAT\SimpleRoster\Security\OAuth\OAuthContext;
-use OAT\SimpleRoster\Security\OAuth\OAuthSigner;
-use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
-use Symfony\Component\Routing\RouterInterface;
 
 class GetUserAssignmentLtiRequestService
 {
-    /** @var OAuthSigner */
-    private $signer;
+    /** @var LtiRequestFactoryInterface */
+    private $ltiRequestFactory;
```

Comment:
Again the same problem with naming consistency. Though `LtiRequestFactory` (the one without version) is not implementing the `LtiRequestFactoryInterface` - looking on the names I would expect a possibility to have it under this variable